code/API_definitions/media-rate-v0.1.yaml

```
@@ -0,0 +1,456 @@
+    - **Three-legged token**: Used for user-authenticated calls. The device is identified from the access token itself, and the `device` object MUST NOT be provided in the request.
+
+    If a two-legged token is used and the device object is missing, the API will return a `422 MISSING_IDENTIFIER` error.
+    If a three-legged token is used and the device object is provided, the API will return a `422 UNNECESSARY_IDENTIFIER` error.
+
+    This approach ensures clarity and security in device identification, and aligns with CAMARA API standards.
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details of how an API consumer requests an access token. Please refer to Identity and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the profile.
+
+    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
+
+    In cases where personal data is processed by the API and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of three-legged access tokens is mandatory. This ensures that the API remains in compliance with privacy regulations, upholding the principles of transparency and user-centric privacy-by-design.
+
+    # Identifying the device from the access token
+
+    This API requires the API consumer to identify a device as the subject of the API as follows:
+    - When the API is invoked using a two-legged access token, the subject will be identified from the device object, which therefore MUST be provided.
+
+    - When a three-legged access token is used however, this optional identifier MUST NOT be provided, as the subject will be uniquely identified from the access token.
+
+    This approach simplifies API usage for API consumers using a three-legged access token to invoke the API by relying on the information that is associated with the access token and was identified during the authentication process.
+
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+externalDocs:
+  description: Product documentation
+  url: https://github.com/camaraproject/MediaRate
+
+servers:
+  - url: "{apiRoot}/media-rate/wip"
+    variables:
+      apiRoot:
+        default: http://localhost:8087
+        description: |
+          API root, defined by service provider, e.g.
+          `api.example.com` or `api.example.com/somepath`
+
+tags:
+  - name: Media Rate
+    description: |
+      Retrieve the maximum media bit rate supported for a given device.
+
+paths:
+  /retrieve-max-media-rate:
+    post:
+      security:
+        - openId:
+            - media-rate:retrieve
```

I would use the full endpoint name here, the logic being that, maybe one day an additional `retrieve` endpoint will be added, and if the scope needed to be renamed, that would be a breaking change. Using the full endpoint name ensures there will be no clashes or ambiguity in future.
```suggestion
            - media-rate:retrieve-max-media-rate
```